signing/secp256k1/suite.go

```diff
@@ -0,0 +1,103 @@
+package secp256k1
+
+import (
+	"crypto/cipher"
+	"crypto/rand"
+
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/btcsuite/btcd/btcec"
+	libp2pCrypto "github.com/libp2p/go-libp2p-core/crypto"
+)
+
+var log = logger.GetOrCreate("crypto/signing/secp256k1")
+
+var _ crypto.Group = (*secp256k1Suite)(nil)
+var _ crypto.Random = (*secp256k1Suite)(nil)
+var _ crypto.Suite = (*secp256k1Suite)(nil)
+
+const Secp256k1 = "secp256k1"
+
+type secp256k1Suite struct{}
+
+// NewSecp256k1 returns a wrapper over ecdsa
+func NewSecp256k1() *secp256k1Suite {
+	return &secp256k1Suite{}
+}
+
+// CreateKeyPair creates a scalar and a point pair that can be used in asymmetric cryptography
+func (s *secp256k1Suite) CreateKeyPair() (crypto.Scalar, crypto.Point) {
+	privKey, publicKey, err := libp2pCrypto.GenerateSecp256k1Key(rand.Reader)
+	if err != nil {
+		panic("could not create secp256k1 key pair: " + err.Error())
+	}
+
+	return &secp256k1Scalar{privKey}, &secp256k1Point{publicKey}
+}
+
+// String returns the string for the group
+func (s *secp256k1Suite) String() string {
+	return Secp256k1
+}
+
+// ScalarLen returns the maximum length of scalars in bytes
+func (s *secp256k1Suite) ScalarLen() int {
+	return btcec.PrivKeyBytesLen
+}
+
+// CreateScalar creates a new Scalar
+func (s *secp256k1Suite) CreateScalar() crypto.Scalar {
+	privKey, _, err := libp2pCrypto.GenerateSecp256k1Key(rand.Reader)
+	if err != nil {
+		panic("could not create secp256k1 key pair: " + err.Error())
+	}
+
+	return &secp256k1Scalar{privKey}
+}
+
+// PointLen returns the max length of point in nb of bytes
+func (s *secp256k1Suite) PointLen() int {
+	return btcec.PubKeyBytesLenCompressed
```

Comment:
is this the Raw representation or another representation?
if it is not the raw then change the marshaling or pointLen as well to match

Reply:
yes, it is the raw representation